src/artifacts_keyring/plugin.py

```diff
@@ -32,7 +33,19 @@ def __init__(self):
             self.exe = [tool_path]
         else:
             try:
-                from dotnetcore2.runtime import get_runtime_path
+                sys_version = tuple(int(i) for i in
+                    subprocess.check_output(["dotnet", "--version"]).decode().strip().partition("-")[0].split("."))
```

Comment:
Coding by exception anti-pattern here, but l guess it simplifies the flow.

Reply:
@aasim If anything, this is the opposite of coding by exception because we're handling too broad a base class. We could have replaced `except Exception` below with `except (subprocess.CalledProcessError, OSError, TypeError, ValueError, OverflowError, UnicodeDecodeError)` to handle all the possibilities, but there's really nothing gained by that.

Also, exception handling in general is not an anti-pattern in Python. It's often more efficient than look-before-you-leap style (unlike say, .NET, which is wildly inefficient).